homeassistant/components/vicare/__init__.py

```diff
@@ -83,12 +81,11 @@ def setup_vicare_api(hass: HomeAssistant, entry: ConfigEntry) -> None:
             "Found device: %s (online: %s)", device.getModel(), str(device.isOnline())
         )
 
-    # Currently we only support a single device
-    device_list = vicare_api.devices
-    device = device_list[0]
-    hass.data[DOMAIN][entry.entry_id][VICARE_DEVICE_CONFIG_LIST] = device_list
-    hass.data[DOMAIN][entry.entry_id][VICARE_DEVICE_CONFIG] = device
-    hass.data[DOMAIN][entry.entry_id][VICARE_API] = getattr(
+    hass.data[DOMAIN][entry.entry_id][VICARE_DEVICE_CONFIG_LIST] = vicare_api.devices
+
+def get_api(entry: ConfigEntry, device):
```

Comment:
```suggestion
def get_api(entry: ConfigEntry, device: PyViCareDeviceConfig) -> Device:
```
Just checked the lib, can we maybe make the code nice with these types?